spec/std/process_spec.cr

```diff
@@ -85,6 +85,79 @@ describe Process do
     $?.exit_code.should eq(0)
   end
 
+  it "getuid" do
+    Process.effective_user_id.should eq `id -u`.to_i
+    Process.effective_user_id.should eq Process.effective_user_id
+  end
+
+  it "getgid" do
+    Process.group_id.should eq `id -g`.to_i
+    Process.group_id.should eq Process.effective_group_id
+  end
+
+  it "become_user" do
+    success = false
+    ruid = Process.user_id
+    euid = Process.effective_user_id
+
+    begin
+      # Leave ruid at 0 to switch back
+      Process.become_user(ruid: 0, euid: 8888, suid: 9999)
+      Process.user_id.should eq 0
+      Process.effective_user_id.should eq 8888
+      success = true
+    rescue ex : Errno
+      # Ok for non-root
+    ensure
+      Process.become_user(ruid: ruid, euid: euid) if success
+    end
+
+    if euid == 0
+      success.should be_true
+    else
+      success.should be_false
+    end
+  end
+
+  it "become_group" do
+    success = false
+    rgid = Process.group_id
+    egid = Process.effective_group_id
+
+    begin
+      Process.become_group(rgid: 7777, egid: 8888, sgid: 9999)
+      Process.group_id.should eq 7777
+      Process.effective_group_id.should eq 8888
+      success = true
+    rescue ex : Errno
+      # Ok for non-root
+    ensure
+      Process.become_group(rgid: rgid, egid: egid) if success
+    end
+
+    if Process.effective_user_id == 0
+      success.should be_true
+    else
+      success.should be_false
+    end
+  end
+
+  it "become_user raises when only setting suid on unsupported platforms" do
+    if !LibC.responds_to?(:setresuid) && LibC.responds_to?(:setreuid)
```

Comment:
I'd move this conditional outside the example, so that it doesn't show up as successful when it actually doesn't do anything.